datastore/redis.go

```diff
@@ -42,11 +47,27 @@ func connectRedis(redisURI string) (*redis.Client, error) {
 	if !strings.HasPrefix(redisURI, "redis://") && !strings.HasPrefix(redisURI, "rediss://") {
 		redisURI = "redis://" + redisURI
 	}
-	opt, err := redis.ParseURL(redisURI)
+
+	redisOpts, err := redis.ParseURL(redisURI)
 	if err != nil {
 		return nil, err
 	}
-	redisClient := redis.NewClient(opt)
+
+	if redisConnectionPoolSize > 0 {
+		redisOpts.PoolSize = redisConnectionPoolSize
+		redisOpts.MinIdleConns = redisConnectionPoolSize / 2
```

Comment:
You might want to add some more information about this value. If the `PoolSize` value is high and the redis service is shared across multiple instances you could end up maxing out the available connections on redis and see random connection errors.

Results from a quick google, but probably better resources out there to explain this:
> Redis nodes can have up to either 10,000 simultaneous connections or 4 simultaneous connections per megabyte of memory, whichever is larger.

Reply:
Good idea. What I could find was 

> In Redis 2.6 and newer, this limit is dynamic: by default it is set to 10000 clients, unless otherwise stated by the maxclients directive in redis.conf.

https://redis.io/docs/reference/clients/

Reply:
maybe it would be better to configure poolSize and minIdleConnections separately 🤔